Deepgram/Clients/Analyze/v1/Client.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright 2021-2024 Deepgram .NET SDK contributors. All Rights Reserved.
+// Use of this source code is governed by a MIT license that can be found in the LICENSE file.
+// SPDX-License-Identifier: MIT
+
+using Deepgram.Models.Analyze.v1;
+using Deepgram.Models.Authenticate.v1;
+
+namespace Deepgram.Clients.Analyze.v1;
+
+/// <summary>
+/// Implements version 1 of the Analyze Client.
+/// </summary>
+/// <param name="apiKey">Required DeepgramApiKey</param>
+/// <param name="deepgramClientOptions"><see cref="DeepgramClientOptions"/> for HttpClient Configuration</param>
+public class Client(string apiKey, DeepgramClientOptions? deepgramClientOptions = null)
+    : AbstractRestClient(apiKey, deepgramClientOptions)
+{
+    #region NoneCallBacks
+    /// <summary>
+    ///  Analyze a file by providing a url 
+    /// </summary>
+    /// <param name="source">Url to the file that is to be analyzed <see cref="UrlSource"></param>
+    /// <param name="analyzeSchema">Options for the transcription <see cref="AnalyzeSchema"/></param>
+    /// <returns><see cref="SyncResponse"/></returns>
+    public async Task<SyncResponse> AnalyzeUrl(UrlSource source, AnalyzeSchema? analyzeSchema, CancellationToken cancellationToken = default)
+    {
+        VerifyNoCallBack(nameof(AnalyzeUrl), analyzeSchema);
+        var stringedOptions = QueryParameterUtil.GetParameters(analyzeSchema);
+        return await PostAsync<SyncResponse>(
+            $"{UriSegments.READ}?{stringedOptions}",
+            RequestContentUtil.CreatePayload(source), cancellationToken);
+    }
+    /// <summary>
+    /// Analyzes a file using the provided byte array
+    /// </summary>
+    /// <param name="source">file is the form of a byte[]</param>
+    /// <param name="analyzeSchema">Options for the transcription <see cref="AnalyzeSchema"/></param>
+    /// <returns><see cref="SyncResponse"/></returns>
+    public async Task<SyncResponse> AnalyzeFile(byte[] source, AnalyzeSchema? analyzeSchema, CancellationToken cancellationToken = default)
+    {
+        VerifyNoCallBack(nameof(AnalyzeFile), analyzeSchema);
+        var stringedOptions = QueryParameterUtil.GetParameters(analyzeSchema);
+        var stream = new MemoryStream();
+        stream.Write(source, 0, source.Length);
+        return await PostAsync<SyncResponse>(
+            $"{UriSegments.READ}?{stringedOptions}",
+            RequestContentUtil.CreateStreamPayload(stream), cancellationToken);
+    }
+
+    /// <summary>
```

Comment:
Will address the TODOs for Logging in these issues since I haven't thought about how to address it yet: https://github.com/deepgram/deepgram-dotnet-sdk/issues/210 and https://github.com/deepgram/deepgram-dotnet-sdk/issues/219